common/model-views.cpp

```diff
@@ -3690,75 +3692,78 @@ namespace rs2
-                        msg << name.first << " (S/N " << name.second << ")\n"
-                            << "Current Version: " << fw << "\n";
-
-                        if (is_rc)
-                            msg << "Release Candidate: " << recommended << " Pre-Release";
-                        else
-                            msg << "Recommended Version: " << recommended;
-
-                        if (!fw_update_required)
-                        {
-                            auto n = std::make_shared<fw_update_notification_model>(
-                                msg.str(), manager, false);
-                            n->delay_id = "dfu." + name.second;
-                            n->enable_complex_dismiss = true;
-                            if (!n->is_delayed())
-                            {
-                                viewer.not_model->add_notification(n);
+                    manager = std::make_shared< firmware_update_manager >( not_model,
+                                                                           *this,
+                                                                           dev,
+                                                                           ctx,
+                                                                           table[product_line],
+                                                                           true );
+                }
 
-                                fw_update_required = true;
+                if( is_upgradeable( fw, recommended ) )
+                {
+                    auto dev_name = get_device_name( dev );
+                    std::stringstream msg;
+                    msg << dev_name.first << " (S/N " << dev_name.second << ")\n"
+                        << "Current Version: " << fw << "\n";
 
-                                related_notifications.push_back(n);
-                            }
-                        }
+                    if( is_rc )
+                        msg << "Release Candidate: " << recommended << " Pre-Release";
+                    else
+                        msg << "Recommended Version: " << recommended;
+
+                    auto n = std::make_shared< fw_update_notification_model >( msg.str(),
+                                                                               manager,
+                                                                               false );
+                    n->delay_id = "dfu." + dev_name.second;
+                    n->enable_complex_dismiss = true;
+                    if( ! n->is_delayed() )
+                    {
+                        not_model->add_notification( n );
+                        related_notifications.push_back( n );
                     }
                 }
             }
         }
+    }
+
+    void device_model::refresh_notifications(viewer_model& viewer)
+    {
+        for (auto&& n : related_notifications) n->dismiss(false);
```

Comment:
All? Not just update notifications?

Reply:
Not my change, it is kept as it was implemented.